src/pykeen/models/predict.py

```diff
@@ -0,0 +1,465 @@
+        ],
+        columns=['head_id', 'head_label', 'score'],
+    ).sort_values('score', ascending=False)
+
+    return _postprocess_prediction_df(
+        rv=rv,
+        add_novelties=add_novelties,
+        remove_known=remove_known,
+        training=model.triples_factory.mapped_triples,
+        testing=testing,
+        query_ids_key='head_id',
+        col=0,
+        other_col_ids=(relation_id, tail_id),
+    )
+
+
+def predict_tails(
+    model: Model,
+    head_label: str,
+    relation_label: str,
+    add_novelties: bool = True,
+    remove_known: bool = False,
+    testing: Optional[torch.LongTensor] = None,
+) -> pd.DataFrame:
+    """Predict tails for the given head and relation (given by label).
+
+    :param head_label: The string label for the head entity
+    :param relation_label: The string label for the relation
+    :param add_novelties: Should the dataframe include a column denoting if the ranked tail entities correspond
+     to novel triples?
+    :param remove_known: Should non-novel triples (those appearing in the training set) be shown with the results?
+     On one hand, this allows you to better assess the goodness of the predictions - you want to see that the
+     non-novel triples generally have higher scores. On the other hand, if you're doing hypothesis generation, they
+     may pose as a distraction. If this is set to True, then non-novel triples will be removed and the column
+     denoting novelty will be excluded, since all remaining triples will be novel. Defaults to false.
+    :param testing: The mapped_triples from the testing triples factory (TriplesFactory.mapped_triples)
+
+    The following example shows that after you train a model on the Nations dataset,
+    you can score all entities w.r.t a given head entity and relation.
+
+    >>> from pykeen.pipeline import pipeline
+    >>> result = pipeline(
+    ...     dataset='Nations',
+    ...     model='RotatE',
+    ... )
+    >>> df = result.model.predict_tails('brazil', 'accusation')
+    """
+    head_id = model.triples_factory.entity_to_id[head_label]
+    relation_id = model.triples_factory.relation_to_id[relation_label]
+    batch = torch.tensor([[head_id, relation_id]], dtype=torch.long, device=model.device)
+    scores = model.predict_scores_all_tails(batch)
+    scores = scores[0, :].tolist()
+    rv = pd.DataFrame(
+        [
+            (entity_id, entity_label, scores[entity_id])
+            for entity_label, entity_id in model.triples_factory.entity_to_id.items()
+        ],
+        columns=['tail_id', 'tail_label', 'score'],
+    ).sort_values('score', ascending=False)
+
+    return _postprocess_prediction_df(
+        rv,
+        add_novelties=add_novelties,
+        remove_known=remove_known,
+        testing=testing,
+        training=model.triples_factory.mapped_triples,
+        query_ids_key='tail_id',
+        col=2,
+        other_col_ids=(head_id, relation_id),
+    )
+
+
+def predict_scores_all_relations(
```

Comment:
from the method names it is not very clear whether the return type is a tensor, or a dataframe.

also sometimes we provide labels, and sometimes ID-tensors.

Reply:
I will clean this up. This will also get another full round of clean up when we make more improvements to id-based triples factories

Reply:
Done in 1bb990da